src/overlay/overlay.scss

```diff
@@ -0,0 +1,379 @@
+// stylelint-disable selector-max-compound-selectors, max-nesting-depth, selector-max-specificity, primer/borders
+// replace with primitive
+$primer-borderRadius-large: 0.75rem;
+
+.Overlay-hidden {
+  display: none !important;
+}
+
+.Overlay-visibilityHidden {
+  height: 0;
+  overflow: hidden;
+  visibility: hidden;
+  opacity: 0;
+}
+
+.Overlay {
+  display: flex;
+  min-width: 192px;
+  flex-direction: column;
+  background-color: var(--color-canvas-default);
+  border-radius: $primer-borderRadius-large;
+  box-shadow: var(--color-overlay-shadow);
+  opacity: 1;
+
+  &.Overlay--height-auto {
+    height: auto;
+  }
+
+  &.Overlay--height-xsmall {
+    height: min(192px, 100% - 2rem);
+  }
+
+  &.Overlay--height-small {
+    height: min(256px, 100% - 2rem);
+  }
+
+  &.Overlay--height-medium {
+    height: min(320px, 100% - 2rem);
+  }
+
+  &.Overlay--height-large {
+    height: min(432px, 100% - 2rem);
+  }
+
+  &.Overlay--height-xlarge {
+    height: min(600px, 100% - 2rem);
+  }
+
+  &.Overlay--width-auto {
+    width: auto;
+  }
+
+  &.Overlay--width-small {
+    width: min(256px, 100% - 2rem);
+  }
+
+  &.Overlay--width-medium {
+    width: min(320px, 100% - 2rem);
+  }
+
+  &.Overlay--width-large {
+    width: min(480px, 100% - 2rem);
+  }
+
+  &.Overlay--width-xlarge {
+    width: min(640px, 100% - 2rem);
+  }
+
+  &.Overlay--width-xxlarge {
+    width: min(960px, 100% - 2rem);
+  }
+
+  &.Overlay--motion-scaleFade {
+    @media screen and (prefers-reduced-motion: no-preference) {
+      animation: 200ms cubic-bezier(0.33, 1, 0.68, 1) 0s 1 normal none running motion--scaleFade;
+    }
+  }
+
+  @keyframes motion--scaleFade {
```

Comment:
It might be safer to prefix the the animation name with the component name: `Overlay--motion-scaleFade`. Since all `@keyframes` are global.